### ISSUE TITLE
Unable to create adx and then use the provider at the same time

### ISSUE THREAD

**Issue by WolfspiritM** [favoretti/terraform-provider-adx#2]
Hi,

When creating a new environment it's not possible to create the ADX Cluster at the same time as using this provider.

The error is:

"failed to get SPT from client credentials: parameter 'resource' cannot be empty"

```
provider "adx" {
  adx_endpoint  = azurerm_kusto_cluster.cluster.uri
  tenant_id = data.azurerm_client_config.current.tenant_id
  client_id       = var.client_id
  client_secret   = var.client_secret
}
```

I think the reason is that the adx_endpoint in this case is empty/not filled. While for resources it can rely on creating the cluster first but the provider is created before anything else.

**Comment by favoretti** [comment]
Hi there and thank you for reporting this. I'm afraid there's little I can do here, since it's internal to how terraform intializes the code. Providers are configured as the first thing, so if there's no cluster yet it can't configure ADX provider :(

**Comment by WolfspiritM** [comment]
Would it be possible to change it so the endpoint can be configured via the resource?

For example the mssql provider we're using aswell to create sql user is setting it like that:
https://github.com/betr-io/terraform-provider-mssql/blob/master/examples/azure/main.tf#L215

**Comment by favoretti** [comment]
Hmmh. That might be an idea. I'll try to look into it this week. Would you be able to compile a provider from a branch to test?

**Comment by favoretti** [comment]
So... Sorry for getting back to this so late. We're currently in progress of adding quite a few more resources to this provider and now I'm thinking about this one. Solution you proposed that mssql provider uses would require us to add auth schema part to each an every resource, defaulting potentially to environment variables. I've heard about a thing called lazy provider initialization, I'll try to read up on that first, before jumping into copypasting code :)

**Comment by favoretti** [comment]
Aha, maybe this is an option:
https://github.com/Mongey/terraform-provider-kafka/issues/178#issuecomment-991178451

**Comment by favoretti** [comment]
Fixed by #8
